android/app/src/main/java/org/fossasia/openevent/viewmodels/UserProfileActivityViewModel.java

```diff
@@ -0,0 +1,67 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+import org.fossasia.openevent.api.APIClient;
+import org.fossasia.openevent.data.auth.User;
+import org.fossasia.openevent.dbutils.LiveRealmDataObject;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.schedulers.Schedulers;
+import timber.log.Timber;
+
+public class UserProfileActivityViewModel extends ViewModel {
+    private CompositeDisposable compositeDisposable;
+    private RealmDataRepository realmRepo;
+    private MutableLiveData<Integer> loadUserFromNetworkResponse;
+    public static final int RETRIEVE_SUCCESSFUL = 1;
+    public static final int RETRIEVE_UNSUCCESSFUL = 2;
+    public static final int RETRIEVE_COMPLETE = 3;
+
+    public UserProfileActivityViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+        compositeDisposable = new CompositeDisposable();
+    }
+
+    public void clearUserDataFromRealm() {
+        realmRepo.clearUserData();
+    }
+    public LiveData<User> getUser() {
+        LiveRealmDataObject<User> liveRealmDataObject = RealmDataRepository.asLiveDataForObject(realmRepo.getUser());
+        Timber.d("User data loaded from disk");
+        return  liveRealmDataObject;
+    }
+    public LiveData<Integer> loadUserFromNetwork(long id) {
+        if(loadUserFromNetworkResponse == null){
+            loadUserFromNetworkResponse = new MutableLiveData<>();
+        }
+        compositeDisposable.add(
+                APIClient.getOpenEventAPI()
+                        .getUser(id)
+                        .subscribeOn(Schedulers.io())
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .subscribe(user -> {
+                                    realmRepo.saveUser(user).subscribe();
+                                    loadUserFromNetworkResponse.setValue(RETRIEVE_SUCCESSFUL);
+                                    Timber.d("User data saved in database");
+                                },
+                                throwable -> {
+                                    loadUserFromNetworkResponse.setValue(RETRIEVE_UNSUCCESSFUL);
+                                    Timber.d(throwable.getMessage() + " Error getting data from network");
+                                },
+                                () -> {
+                                    loadUserFromNetworkResponse.setValue(RETRIEVE_COMPLETE);
+                                    Timber.d("User data loaded from network");
+                                })
+        );
```

Comment:
There's too much nesting. Move api code above and save the disposable in a variable and add in composite disposable below